js/compiler.ts

```diff
@@ -55,6 +55,10 @@ type OutputCode = string;
  * The original source code
  */
 type SourceCode = string;
+/**
+ * Flags that are passed from CLI arguments
+ */
+type CompilerFlags = { [key: string]: boolean };
```

Comment:
@ry I guess we will have to replace `boolean` with `any` pretty soon, but it should be enough for now.

Reply:
@bartlomieju I think these are TS compiler flags - so recompile shouldn’t go in there... but maybe @kitsonk can suggest the best place to put this flag.

Reply:
Sorry I misunderstood this when reviewing it the first time. I see that you're adding this now.

I'd prefer if you did not generalize - I think a simple `recompile: bool` member would work. When there are more than 3 compiler flags we can discuss generalizing.